### PR TITLE
Updated the way scrum answers are assigned

### PR DESCRIPTION
**Pull request by gouravagrwal**
Updated the way in which scrum answers are being assigned in scrum report. Fixes issue #127.

**Files changed:** 
 **1. ScrumReportRepository.cs** - Modified the way scrum answers are assigned for each employee and changed the mehods to use any() instead of count()
 **2. ScrumReportRepositoryTest.cs** - Made changes as per the changes in the way answers are assigned and updated test cases to use any() instead of count()
 **3. scrumProject-Details.html** - Changed the view as per the changes in the way answers for scrum are being assigned
**4. LeaveReportRepository.cs** - Updated methods to use Any() instead of Count()
**5. LeaveReportRepositoryTest.cs** - Updated test cases to use Any() instead of count()
**6. IStringConstantRepository.cs** - Added a constant string
**7. StringConstantRepository.cs** - Added a constant string.

### CODE REVIEW

File: Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs [hunk 1]
```diff
@@ -81,7 +81,7 @@ public ScrumReportRepository(IRepository<Scrum> scrumDataRepository, IRepository
             //Fetch all the scrum answers for a particular employee
             List<ScrumAnswer> scrumAnswers = (await _scrumAnswerDataRepository.FetchAsync(x => x.EmployeeId == user.Id)).ToList();
             //Find scrum answers for a particular employee of a particular project on a specific date 
-            List<ScrumAnswer> todayScrumAnswers = scrumAnswers.FindAll(x => x.AnswerDate == scrumDate && x.ScrumId == scrum.Id).ToList();
+            List<ScrumAnswer> todayScrumAnswers = scrumAnswers.FindAll(x => x.AnswerDate.Date == scrumDate.Date && x.ScrumId == scrum.Id).ToList();
             employeeScrumDetail.EmployeeName = string.Format("{0} {1}", user.FirstName, user.LastName);
             //Assigning answers to specific scrum questions
             if (todayScrumAnswers.Count() == 0)
```
Comment (chintans) [inline Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs]: Use Any() and not Count()

File: Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs [hunk 2]
```diff
@@ -90,22 +90,32 @@ public ScrumReportRepository(IRepository<Scrum> scrumDataRepository, IRepository
             }
             foreach (var todayScrumAnswer in todayScrumAnswers)
             {
-                if (todayScrumAnswer.Question.QuestionStatement.Equals(_stringConstant.ScrumFirstQuestion))
+                if (todayScrumAnswer.Question.Type == 1 && todayScrumAnswer.Question.OrderNumber == 1)
```
Comment (chintans) [inline Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs:93]: Not sure what 1 means here?

File: Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs [hunk 3]
```diff
@@ -90,22 +90,32 @@ public ScrumReportRepository(IRepository<Scrum> scrumDataRepository, IRepository
...
             }
             return employeeScrumDetail;
         }
 
+        /// <summary>
+        /// Method to split the multiple scrum answers into an array
+        /// </summary>
+        /// <param name="Answer"></param>
+        /// <returns>scrum answer</returns>
+        private string[] SplitAnswer(string Answer)
```
Comment (chintans) [inline Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs]: naming conventions are wrong, for just splitting a string, why have you created a separate method?

**Review by chintans — CHANGES_REQUESTED** [review]

File: Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs (hunk 2 above)
Comment (gouravagrwal) [reply Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs:93]: Earlier, it was used to differentiate between scrum and task mail questions.  Also, the order number that follows in the condition was used to differentiate between the various scrum questions. Now, these have been converted to enums and the respective changes have been made which will be updated after the pull request #122 is merged.

File: Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs (hunk 3 above)
Comment (gouravagrwal) [reply Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs]: This method is created to split the scrum answers that are stored in the form of a single string into an array of strings. This splitting of answers is used at several places, so to reduce redundancy I created a separate method for it.

File: Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs (hunk 1 above)
Comment (gouravagrwal) [reply Slack.Automation/Promact.Core.Repository/ScrumReportRepository/ScrumReportRepository.cs]: I have updated the code as per your feedbacks.

**Review by chintans — APPROVED** [review]

**Pull request merged**